### PR TITLE
Convert UntypedActor to AbstractActor

### PR DESCRIPTION
**Pull request by ortibazar**
Refs #22308

### CODE REVIEW

**Comment by patriknw** [comment]
PLS WHITELIST

File: akka-docs/rst/java/code/docs/cluster/TransformationBackend.java [hunk 1]
```diff
@@ -3,58 +3,54 @@
...
+        register(mUp.member());
+      })
+      .build();
   }
 
   void register(Member member) {
     if (member.hasRole("frontend"))
       getContext().actorSelection(member.address() + "/user/frontend").tell(
-          BACKEND_REGISTRATION, getSelf());
+          BACKEND_REGISTRATION, self());
```
Comment (patriknw) [inline akka-docs/rst/java/code/docs/cluster/TransformationBackend.java]: good attention to details :)

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 2]
```diff
@@ -167,7 +167,7 @@ New::
   }
 
 It's recommended to migrate ``UntypedActor`` to ``AbstractActor`` by implementing
-``createReceive`` instead of ``onMessage``.
+``createReceive`` instead of ``onReceive``.
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: good catch

**Review by patriknw — APPROVED** [review]
LGTM, thanks so much

**Comment by patriknw** [comment]
PLS BUILD

File: akka-docs/rst/java/code/docs/cluster/FactorialBackend.java [hunk 3]
```diff
@@ -1,38 +1,33 @@
...
-          return factorial(n);
-        }
-      }, getContext().dispatcher());
-
-      Future<FactorialResult> result = f.map(
+  public Receive createReceive() {
+    return receiveBuilder()
+      .match(Integer.class, n -> {
+        Future<BigInteger> f = future(() -> factorial(n),
+          getContext().dispatcher());
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/cluster/FactorialBackend.java]: Could we make it `CompletionStage` instead?

File: akka-docs/rst/java/code/docs/cluster/FactorialBackend.java [hunk 4]
```diff
@@ -1,38 +1,33 @@
...
+
+        Future<FactorialResult> result = f.map(
           new Mapper<BigInteger, FactorialResult>() {
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/cluster/FactorialBackend.java]: And skip the old `mapper` and use the built in transform method in completion stage instead (can't remember which one of the 500 methods it is from the top of my head though)?

**Review by johanandren — APPROVED** [review]
LGTM, great work! Added some notes but we could make those a separate ticket if you are not up for it!

File: akka-docs/rst/java/code/docs/cluster/FactorialBackend.java (hunk 3 above)
Comment (patriknw) [inline akka-docs/rst/java/code/docs/cluster/FactorialBackend.java]: I have created a ticket for this https://github.com/akka/akka/issues/22393

**Pull request merged**